### DISCUSSION TITLE
Performance rating Tool

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#50] [Ideas]
I checked the code and have few proposal:
- suppress the 10x run and calculate the average 1x from the 100x run.
- make a high level microsecond counter masking the PIT or HPET usage. One function could reveal the used mechanism.
- try to avoid the double function call while testing speed.
- why adding 100x the time of 1x instead of counting 1x the time of 100x ?

**Comment by VendelinSlezak** [comment]
I added these things to Update 10.

**Comment by defdefred** [comment]
New PR.
Seems that redraw_part_of_screen                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            is not testing screen overflow.
Are we sure that testing is not needed?


timing is not consistent, shouldn't we masking interrupt while running task?

**Comment by VendelinSlezak** [comment]
> Seems that redraw_part_of_screen is not testing screen overflow.

It is testing overflow:

>  if(x>graphic_screen_x || y>graphic_screen_y) {
>   return;
>  }
>  if((x+width)>graphic_screen_x) {
>   width = (graphic_screen_x-x);
>  }
>  if((y+height)>graphic_screen_y) {
>   height = (graphic_screen_y-y);
>  }

> timing is not consistent, shouldn't we masking interrupt while running task?

In emulators it is impossible to get consistent timing, because host operating system runs multiple other tasks beside emulator, but I tested Performance rating on real computers and it gives very consistent results, only few microseconds up/down what is fine.

**Comment by defdefred** [comment]
mmmm try "redraw 25% of screen 50% right visible"

**Comment by VendelinSlezak** [comment]
You probably improved code from this comment, but I noticed that you are using redraw_part_of_framebuffer function instead of redraw_part_of_screen.
